---
title: "recount2_data"
output: html_notebook
---

```{r}
library('recount')
library('DESeq2')
library('regionReport')
```

Say we already have a project of interest, possibly because we searched for it in the [Shiny interface](https://jhubiostatistics.shinyapps.io/recount/).  Let's download it and load its gene expression summary:

```{r}
project_info <- abstract_search('GSE32465')
download_study(project_info$project)
load(file.path(project_info$project, 'rse_gene.Rdata'))
```

recount can open a link to the SRA metadata page for the study, which can be helpful:

```{r}
browse_study(project_info$project)
```

```{r}
## Extract the sample characteristics
geochar <- lapply(split(colData(rse_gene), seq_len(nrow(colData(rse_gene)))), geo_characteristics)
```

This study requires a little massaging to get metadata expressed in terms we can use:

```{r}
## Note that the information for this study is a little inconsistent, so we
## have to fix it.
geochar <- do.call(rbind, lapply(geochar, function(x) {
    if('cells' %in% colnames(x)) {
        colnames(x)[colnames(x) == 'cells'] <- 'cell.line'
        return(x)
    } else {
        return(x)
    }
}))

## We can now define some sample information to use
sample_info <- data.frame(
    run = colData(rse_gene)$run,
    group = ifelse(grepl('uninduced', colData(rse_gene)$title), 'uninduced', 'induced'),
    gene_target = sapply(colData(rse_gene)$title, function(x) { strsplit(strsplit(x,
        'targeting ')[[1]][2], ' gene')[[1]][1] }),
    cell.line = geochar$cell.line
)
```

We apply some mild normalization to account for varying sequencing depth across samples:

```{r}
rse <- scale_counts(rse_gene)
```

We augment the metadata with the variables we want to use for differential expression analysis:

```{r}
colData(rse)$group <- sample_info$group
colData(rse)$gene_target <- sample_info$gene_target
```

Use DESeq2 to perform differential expression analysis

```{r}
dds <- DESeqDataSet(rse, ~ gene_target + group)
dds <- DESeq(dds, test = 'LRT', reduced = ~ gene_target, fitType = 'local')
res <- results(dds)
```

## Explore results
plotMA(res, main="DESeq2 results for SRP009615")

## Make a report with the results
DESeq2Report(dds, res = res, project = 'SRP009615',
    intgroup = c('group', 'gene_target'), outdir = '.',
    output = 'SRP009615-results')
```
